src/components/ManageHiddenAssets/ManageHiddenAssetsList.tsx

```diff
@@ -0,0 +1,171 @@
+import {
+  Box,
+  Flex,
+  IconButton,
+  Menu,
+  MenuButton,
+  MenuItem,
+  MenuList,
+  Stack,
+  Text,
+  VStack,
+} from '@chakra-ui/react'
+import type { AssetId } from '@shapeshiftoss/caip'
+import { fromAssetId, isNft } from '@shapeshiftoss/caip'
+import { isToken } from '@shapeshiftoss/utils'
+import { useCallback, useMemo } from 'react'
+import { TbDots, TbExternalLink, TbEye, TbFileText } from 'react-icons/tb'
+import { useTranslate } from 'react-polyglot'
+import { useNavigate } from 'react-router-dom'
+
+import { Amount } from '@/components/Amount/Amount'
+import { AssetIcon } from '@/components/AssetIcon'
+import { RawText } from '@/components/Text'
+import { isSome } from '@/lib/utils'
+import { preferences } from '@/state/slices/preferencesSlice/preferencesSlice'
+import {
+  selectAssetById,
+  selectAssets,
+  selectPortfolioCryptoPrecisionBalanceByFilter,
+} from '@/state/slices/selectors'
+import { useAppDispatch, useAppSelector } from '@/state/store'
+
+const dotsIcon = <TbDots />
+const eyeIcon = <TbEye />
+const externalLinkIcon = <TbExternalLink />
+const fileTextIcon = <TbFileText />
+
+const hoverStylesSx = { _hover: { bg: 'gray.50', _dark: { bg: 'gray.700' } } }
+
+type ManageHiddenAssetsListProps = {
+  onClose?: () => void
+}
+
+export const ManageHiddenAssetsList: React.FC<ManageHiddenAssetsListProps> = ({ onClose }) => {
+  const translate = useTranslate()
+  const navigate = useNavigate()
+  const appDispatch = useAppDispatch()
+
+  const spamMarkedAssetIds = useAppSelector(preferences.selectors.selectSpamMarkedAssetIds)
+
+  const hiddenAssets = useAppSelector(state =>
+    spamMarkedAssetIds.map(assetId => selectAssetById(state, assetId)).filter(isSome),
+  )
+
+  const assets = useAppSelector(selectAssets)
+
+  const getExplorerHref = (assetId: AssetId) => {
+    const asset = assets[assetId]
+    if (!asset) return
+
+    const { assetReference } = fromAssetId(assetId)
+
+    if (isNft(assetId)) {
+      const [token] = assetReference.split('/')
+      return `${asset.explorer}/token/${token}?a=${asset.id}`
+    }
+
+    if (isToken(assetId)) return `${asset?.explorerAddressLink}${assetReference}`
+
+    return asset.explorer
+  }
```

Comment:
memoize me baby

Reply:
It could even be in the explorerHref memo directly without the need for a function